### ISSUE TITLE
Reopened https://github.com/kivy/kivy/issues/5784

### ISSUE THREAD

**Issue by kcrossen** [kivy/kivy#6195]
### Versions

* Python: 2.7.16
* OS: 10.11.6
* Kivy: Kivy-1.10.1_osx10.13.5_builtin_python2_r1.dmg
* Kivy installation method: Every one that you've documented

### Description
level=2
```
  File "pythonforandroid/build.py", line 540, in build_recipes
  File "pythonforandroid/recipe.py", line 293, in download_if_necessary
  File "pythonforandroid/recipe.py", line 339, in download
  File "pythonforandroid/recipe.py", line 154, in download_file
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib.py", line 245, in retrieve
    fp = self.open(url, data)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib.py", line 213, in open
    return getattr(self, name)(url)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/urllib.py", line 443, in open_https
    h.endheaders(data)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1038, in endheaders
    self._send_output(message_body)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 882, in _send_output
    self.send(msg)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 844, in send
    self.connect()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/httplib.py", line 1263, in connect
    server_hostname=server_hostname)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 369, in wrap_socket
    _context=self)
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 599, in __init__
    self.do_handshake()
  File "/Library/Frameworks/Python.framework/Versions/2.7/lib/python2.7/ssl.py", line 828, in do_handshake
    self._sslobj.do_handshake()
IOError: [Errno socket error] [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed (_ssl.c:727)
```

### Code and Logs
Ugly temporary fix in ssl.py starting at line 590:
```
            # create the SSL object
            try:
                self._context.check_hostname = False
                self._context.verify_mode = CERT_NONE
                self._sslobj = self._context._wrap_socket(self._sock, server_side,
                                                          server_hostname, ssl_sock=self)
                if do_handshake_on_connect:
```

**Comment by tito** [comment]
First, this is NOT a kivy issue, but a python-for-android issue, please use the right issue tracker.
Second, you cropped the log, this strip important information. We can't tell if it's an issue with your system, or an issue with the file you're currently downloading because we don't know what it it.

Please opening issues, try to talk first on support channels.

**Comment by kcrossen** [comment]
I've already addressed this in #general anyway. It's a systemic issue that will affect most "naked" buildozer builds on any OSX version that doesn't match the OSX et-al used by the author/tester of the script. OSX is a bit of a mess Python-wise, I believe, and the "naked" buildozer script does not download standard "modules" for any "modules" it thinks it has found on the host system. For example, if certifi is not present, kablooey. There's no practical way to check everything on the OS host, so unless the script developer assumes a "virgin" OS host, kablooey is practically guaranteed.

It took some time to figure out what happened here, but I wound up posting the simplest possible workaround when I realized how many other similar problems awaited.
